rs/sns/testing/README.md

```diff
@@ -44,14 +43,183 @@ To run the scenario on the local PocketIC instance:
+<br>
+
+At this point we assume that SNS named "Daniel" was created and its swap was successfully completed.
+
+1) Get SNS neuron ID contolled by `sns-testing` identity:
+   ```
+   IC_URL="http://127.0.0.1:8080" quill sns neuron-id --principal-id "$(dfx identity get-principal --identity sns-testing)" --memo 42
+   ```
+
+   ```
+   SNS Neuron Id: a96c889f2eab3fb4ae7aac3978f04eeb039e0ec8047516fcd8fae8b20bd75502
+   ```
+
+2) Prepare `sns_canister_ids.json`
+
+   Get SNS canister IDs:
+   ```
+   IC_URL="http://127.0.0.1:8080" quill --insecure-local-dev-mode sns list-deployed-snses
+   ```
+
+   ```
+   cat >> sns_canister_ids.json<< EOF
+   {
+      "root_canister_id":"7tjcv-pp777-77776-qaaaa-cai",
+      "governance_canister_id":"7uieb-cx777-77776-qaaaq-cai",
+      "index_canister_id":"7pnye-yp777-77776-qaaca-cai",
+      "swap_canister_id":"72kjj-zh777-77776-qaabq-cai",
+      "ledger_canister_id":"75lp5-u7777-77776-qaaba-cai"
+   }
+   EOF
+   ```
+
+   SNS canister IDs may vary for you.
+
+3) Prepare quill `message.json` with SNS-controlled canister upgrade proposal:
+   ```
+   quill sns --pem-file ~/.config/dfx/identity/sns-testing/identity.pem --canister-ids-file sns_canister_ids.json make-upgrade-canister-proposal \
+      --target-canister-id lxzze-o7777-77777-aaaaa-cai --wasm-path "$(bazel info bazel-bin)/rs/sns/testing/sns_testing_canister.wasm.gz" \
+      --title "Upgrade SNS-controlled-canister" --mode upgrade "<Neuron ID>" > message.json
+   ```
+
+4) Submit the message with SNS proposal:
+   ```
+   IC_URL="http://127.0.0.1:8080" quill --insecure-local-dev-mode send message.json
+   ```
+
+   This command will return the ID of the newly created proposal
+   ```
+   ...
+   Successfully created new proposal with ID 1
+   ```
+
+</details>
+
+Once the SNS proposal is created, the voting begins.
+To upvote the proposal using Neurons controlled by identities that participated in the swap on the previous step run:
+```
+bazel run //rs/sns/testing:sns-testing -- --network http://127.0.0.1:8080 sns-proposal-upvote --sns-name "<SNS name>" --proposal-id "<Proposal ID>" --wait
+```
+
+## Get ICP tokens
+
+You can use `bazel run //rs/sns/testing:sns-testing -- --network http://127.0.0.1:8080 transfer-icp` to get ICP tokens to your account.
+
+This subcommand supports transfer to the direct account ID, e.g.:
+```
+bazel run //rs/sns/testing:sns-testing -- --network http://127.0.0.1:8080 transfer-icp --amount 650.0 --to 5c9b28f3e2218975ea76449cf9b949a860d741a3af7dd4dd062f7481e3a99cde
+```
+
+Or to the principal by its ID (and optionally subaccount):
+```
+bazel run //rs/sns/testing:sns-testing -- --network http://127.0.0.1:8080 transfer-icp --amount 650.0 --to-principal fomoo-4i5fe-epl6n-dc7hi-lfqhi-4ii4q-txsav-xvenw-ffacw-youhl-mae
+```
+
+Obtained ICP tokens can be used to participate in SNS swap or you may stake them to get NNS neuron. This can be done either using `quill` or NNS dapp web UI.
```

Comment:
Also, ICP is needed to buy cycles that are required for some SNS operations, e.g., upgrading dapps with large Wasms